components/dashboard/src/settings/Plans.tsx

```diff
@@ -280,6 +268,29 @@ export default function () {
             setConfirmDowngradeToPlan(undefined);
         }
     }
+    const confirmCancelDowngrade = () => {
+        if (!scheduledDowngradePlanId) {
+            // No scheduled downgrade to be cancelled?
+            return;
+        }
+        if ((paidSubscription?.paymentReference || '').startsWith("github:")) {
+            return;
+        }
+        setIsConfirmCancelDowngrade(true);
+    }
+    const doCancelDowngrade = async (event: React.MouseEvent) => {
+        if (!isConfirmCancelDowngrade || !paidSubscription) {
+            return;
+        }
+        try {
+            await server.subscriptionCancelDowngrade(paidSubscription.uid);
+            // TODO: Update state somehow?
```

Comment:
Maybe we want to optimistically delete `scheduledDowngradePlanId` here, in order to stop showing the just-cancelled downgrade (immediately, as there is probably no need to wait for the Chargebee confirmation here).

Reply:
Fixed by introducing a `pendingDowngradeCancellation`.